app/javascript/components/reconfigure-vm-form/index.jsx

```diff
@@ -0,0 +1,270 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Loading } from 'carbon-components-react';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import { createSchema } from './reconfigure-form.schema';
+import { CellAction } from '../miq-data-table/helper';
+import DiskTable from './disk-table';
+import NetworkTable from './network-table';
+import CdDriveTable from './cd-drive-table';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import {
+  setInitialData,
+  dataTableHeaders,
+  subFormSubmit,
+  setButtonActions,
+  reconfigureSubmitData,
+  validateMemory,
+  getFormSubmitStatus,
+  validateCpu,
+} from './helper';
+import { getFormsData } from './helpers/general';
+import { setDiskData, validateDiskSize } from './helpers/disk';
+import { setNetworkData, setNetworkOptions } from './helpers/network';
+import { setDrivesData } from './helpers/drive';
+
+const ReconfigureVmForm = ({
+  recordId, requestId, roles, options, memory,
+}) => {
+  const formData = getFormsData();
+  const [data, setData] = useState({
+    initialValues: {},
+    isLoading: true,
+    form: {
+      title: formData.reconfigure.title,
+      type: 'reconfigure',
```

Comment:
could you use this like enums `TYPE.RECONFIGURE`
```
const TYPE = {
    RECONFIGURE: 'reconfigure',
  };
```

Reply:
done